From 28af5ac996c94922fc49660e0efa955f57575187 Mon Sep 17 00:00:00 2001
From: David Rheinsberg <david.rheinsberg@gmail.com>
Date: Mon, 21 Jun 2021 11:12:41 +0200
Subject: [PATCH] units: explicitly order against dbus.socket and
 sysinit.target

First of all, order all our units after dbus.socket. This is implicit
in systemd for all socket activated units. However, dues to the aliasing
to dbus.service, this is only true if our units are enabled. Hence, by
being explicit, we make sure the dependency is correctly shown when
debugging disabled services as well.

Secondly, make sure to order after sysinit.target, so basic system
setup is done before we start. This is especially important to make sure
the launcher can resolve user names and read the disk.

Signed-off-by: David Rheinsberg <david.rheinsberg@gmail.com>
---
 src/units/system/dbus-broker.service.in | 1 +
 src/units/user/dbus-broker.service.in   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/units/system/dbus-broker.service.in b/src/units/system/dbus-broker.service.in
index 4155630..37d5dd7 100644
--- a/src/units/system/dbus-broker.service.in
+++ b/src/units/system/dbus-broker.service.in
@@ -2,6 +2,7 @@
 Description=D-Bus System Message Bus
 Documentation=man:dbus-broker-launch(1)
 DefaultDependencies=false
+After=dbus.socket
 Before=basic.target shutdown.target
 Requires=dbus.socket
 Conflicts=shutdown.target
diff --git a/src/units/user/dbus-broker.service.in b/src/units/user/dbus-broker.service.in
index 9580aab..cd8d5c0 100644
--- a/src/units/user/dbus-broker.service.in
+++ b/src/units/user/dbus-broker.service.in
@@ -2,6 +2,7 @@
 Description=D-Bus User Message Bus
 Documentation=man:dbus-broker-launch(1)
 DefaultDependencies=false
+After=dbus.socket
 Before=basic.target shutdown.target
 Requires=dbus.socket
 Conflicts=shutdown.target
